src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,1055 @@
+                    updateLatestControlTimestamp();
+                    break;
+                case "Reboot":
+                    reboot();
+                    break;
+                default:
+                    break;
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+    }
+
+    @Override
+    public void controlProperties(List<ControllableProperty> list) throws Exception {
+        if (CollectionUtils.isEmpty(list)) {
+            throw new IllegalArgumentException("Controllable properties cannot be null or empty");
+        }
+
+        for (ControllableProperty controllableProperty : list) {
+            controlProperty(controllableProperty);
+        }
+    }
+
+    /**
+     * Interceptor for RestTemplate that injects
+     * Cookie header and reacts to the communicator state - after reboot the certificates on the device get
+     * refreshed that results with timeouts on https requests. In order to overcome this - communicator is being internally
+     * destroyed on the successful reboot action. Hovewer, we cannot guarantee that any other resource does not issue
+     * reboot or certificates do not get refreshed for other reasons, so we check for a specific error on
+     * authorization event and reinitialize the communicator here.
+     */
+    class PolycomVideoOSInterceptor implements ClientHttpRequestInterceptor {
+        @Override
+        public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttpRequestExecution execution) throws IOException {
+            if (!isInitialized()) {
+                try {
+                    internalInit();
+                } catch (Exception e) {
+                    throw new IOException(String.format("Unable to request %s: device is not initialized: %s", request.getURI(), e.getMessage()));
+                }
+            }
+
+            ClientHttpResponse response = execution.execute(request, body);
+            if (response.getRawStatusCode() == 403) {
+                try {
+                    authenticate();
+                    response = execution.execute(request, body);
+                } catch (Exception e) {
+                    logger.error("Authentication failed during interception: " + e.getMessage());
+                    if (e.getMessage().startsWith("Cannot reach resource") && e.getMessage().contains(SESSION)) {
+                        // In case it's been rebooted by some other resource - we need to react to that.
+                        // Normally we expect that the authorization request timeouts, since previous one failed with
+                        // a specific error code, so basically we do the same as before - authenticate + retry previous
+                        // request but with the destroy action called first, since init will be done next when
+                        // authentication is requested.
+                        internalDestroy();

```

Comment:
The problem is that if the device gets rebooted - the previously accepted certificates won't work anymore, but it may be not only the adapter who initiates the reboot, but it may be done externally which would lead to the same "Cannot reach resource" error. 
We have discussed this solution with Shervin, but maybe we can go with a more elegant one if similar issue appeared in the past

Reply:
This is not logical. If there is a need to re-authenticate, it should be triggered by certain error. As for "previously accepted certificates won't work anymore" - I do not understand what it means. This is HTTP communicator, there is no permanent connection, remote side will keep HTTP connection alive for a bit (if it supports HTTP 1.1), but if it is rebooted as you said of course new connection will be established and each new connection starts with SSL handshake which involves certificate exchange, etc. I do not understand what it is trying to resolve here and why object has to be fully destroyed in order to resolve that.

Reply:
Ok I guess I need to investigate this then, when the devices are back (Had a chat with Connor about this and he said it might be a bug or something, I will make a doublecheck with the new firmware also). 

I understand that the new connection is established, and the certificate exchange should be performed again but from what I have experienced - each time after device reboot, all url calls made by the communicator would throw "Cannot reach resource" indefinitely, unless communicator is re-initialized.

Reply:
Did you try to call disconnect() instead? It will destroy the http client itself, but will not require to re-initialize the whole communicator object. And http client will be re-created when next request is processed by HttpCommunicator

Reply:
I don't remember if I did, so I will do this and double check. 
I do remember just calling internalInit() without having it destroyed, which also worked, so I have decided to stick to destroy after some discussions, as it would allow to stick to the init state of the adapter for the re-initialization.